### PR TITLE
Clarifying documentation for Cones

### LINKED ISSUE #2891: Improve docstrings for Cones (and possibly others)
This pertains to the Cone docstrings: https://docs.openmc.org/en/stable/pythonapi/generated/openmc.Cone.html, and others.

The documentation for `openmc.Cone` lacks a quadric surface equation, and more importantly, an intuitive explanation of `r2`. `[X|Y|Z]Cone` does include the equation, but `r2` is still not super intuitive.

In general many surfaces' documentation probably needs more intuitive explanations. 

Thanks @pshriwise for making me find this.

### PR DESCRIPTION
**Pull request by MicahGale**
# Description

This provides some clarification on how `Cone` surfaces are formed, to address some user confusion that occured. 

Fixes #2891.

# Checklist

- [x] I have performed a self-review of my own code
- [ ] ~I have run [clang-format](https://docs.openmc.org/en/latest/devguide/styleguide.html#automatic-formatting) (version 15) on any C++ source files (if applicable)~
- [x] I have followed the [style guidelines](https://docs.openmc.org/en/latest/devguide/styleguide.html#python) for Python source files (if applicable)
- [x] I have made corresponding changes to the documentation (if applicable)
- [ ] ~I have added tests that prove my fix is effective or that my feature works (if applicable)~

### CODE REVIEW

File: openmc/surface.py [hunk 1]
```diff
@@ -1757,7 +1762,9 @@ class Cone(QuadricMixin, Surface):
     z0 : float, optional
         z-coordinate of the apex in [cm]. Defaults to 0.
     r2 : float, optional
-        Parameter related to the aperature. Defaults to 1.
+        Parameter related to the aperature [cm^2]
+        This is the square of the radius of the cone 1 cm from.
```
[the remark this answers is not in the document]
Comment (gridley) [reply openmc/surface.py:1766]: Incorrect use of square brackets for units:

https://hsm.stackexchange.com/questions/5907/old-square-bracket-notation-for-units

File: openmc/surface.py [hunk 2]
```diff
@@ -1900,6 +1907,10 @@ class XCone(QuadricMixin, Surface):
     """A cone parallel to the x-axis of the form :math:`(y - y_0)^2 + (z - z_0)^2 =
     r^2 (x - x_0)^2`.
 
+    .. Note::
+        This creates a double cone, which is two "traditional" cones that meet at their apex.
+        For a "traditional" cone see :class:`~openmc.model.XConeSided`.
```
[the remark this answers is not in the document]
Comment (gridley) [reply openmc/surface.py:1912]: https://hsm.stackexchange.com/questions/5907/old-square-bracket-notation-for-units

**Review by gridley — CHANGES_REQUESTED** [review]

File: openmc/surface.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MicahGale) [reply openmc/surface.py:1912]: @gridley how is this relevant to these lines?

File: openmc/surface.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MicahGale) [reply openmc/surface.py:1766]: Square brackets are used throughout the documentation for units, and is beyond the scope of this PR. 

In general I disagree with your sentiment about this notation. Also "Only the Sith deal in absolutes".

File: openmc/surface.py [hunk 3]
```diff
@@ -2110,8 +2110,8 @@ class ZCone(QuadricMixin, Surface):
     r^2 (z - z_0)^2`.
 
     .. Note::
-        This creates a double cone, which is two "traditional" cones that meet at their apex.
-        For a "traditional" cone see :class:`~openmc.model.ZConeSided`.
+        This creates a double cone, which is two one-sided cones that meet at their apex.
+        For a one-sided cone see :class:`~openmc.model.ZConeSided`.
```
Comment (pshriwise) [inline openmc/surface.py:2114]:
```suggestion
        For a one-sided cone see :class:`~openmc.model.ZConeOneSided`.
```

File: openmc/surface.py [hunk 4]
```diff
@@ -2009,8 +2009,8 @@ class YCone(QuadricMixin, Surface):
     r^2 (y - y_0)^2`.
 
     .. Note::
-        This creates a double cone, which is two "traditional" cones that meet at their apex.
-        For a "traditional" cone see :class:`~openmc.model.YConeSided`.
+        This creates a double cone, which is two one-sided cones that meet at their apex.
+        For a one-sided cone see :class:`~openmc.model.YConeSided`.
```
Comment (pshriwise) [inline openmc/surface.py:2013]:
```suggestion
        For a one-sided cone see :class:`~openmc.model.YConeOneSided`.
```

File: openmc/surface.py [hunk 5]
```diff
@@ -1908,8 +1908,8 @@ class XCone(QuadricMixin, Surface):
     r^2 (x - x_0)^2`.
 
     .. Note::
-        This creates a double cone, which is two "traditional" cones that meet at their apex.
-        For a "traditional" cone see :class:`~openmc.model.XConeSided`.
+        This creates a double cone, which is two one-sided cones that meet at their apex.
+        For a one-sided cone see :class:`~openmc.model.XConeSided`.
```
Comment (pshriwise) [inline openmc/surface.py:1912]:
```suggestion
        For a one-sided cone see :class:`~openmc.model.XConeOneSided`.
```

File: openmc/surface.py [hunk 6]
```diff
@@ -1749,8 +1749,8 @@ class Cone(QuadricMixin, Surface):
     """A conical surface parallel to the x-, y-, or z-axis.
 
     .. Note::
-        This creates a double cone, which is two "traditional" cones that meet at their apex.
-        For a "traditional" cone see :class:`~openmc.model.XConeSided`,
+        This creates a double cone, which is two one-sides cones that meet at their apex.
+        For a one-sided cone see :class:`~openmc.model.XConeSided`,
```
Comment (pshriwise) [inline openmc/surface.py:1753]:
```suggestion
        For a one-sided cone see :class:`~openmc.model.XConeOneSided`,
```

File: openmc/surface.py [hunk 7]
```diff
@@ -1749,8 +1749,8 @@ class Cone(QuadricMixin, Surface):
     """A conical surface parallel to the x-, y-, or z-axis.
 
     .. Note::
-        This creates a double cone, which is two "traditional" cones that meet at their apex.
-        For a "traditional" cone see :class:`~openmc.model.XConeSided`,
+        This creates a double cone, which is two one-sides cones that meet at their apex.
```
Comment (pshriwise) [inline openmc/surface.py:1752]:
```suggestion
        This creates a double cone, which is two one-sided cones that meet at their apex.
```

**Review by pshriwise — CHANGES_REQUESTED** [review]
Some class name corrections and a typo fix to tackle but then I think this looks good!

File: openmc/surface.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MicahGale) [reply openmc/surface.py]: Corrected all class names, and verified that the link is correct.

**Comment by MicahGale** [comment]
I came across this documentation:

> A one-sided cone is composed of a normal cone surface and an “ambiguity” surface that eliminates the ambiguity as to which region of space is included. This class acts as a proper surface, meaning that unary + and - operators applied to it will produce a half-space. The negative side is defined to be the region inside of the cone.

I feel like "ambiguity" surface here is a bit counterintuitive. Would "disambiguation" surface be better?

**Comment by pshriwise** [comment]
> I came across this documentation:
> 
> > A one-sided cone is composed of a normal cone surface and an “ambiguity” surface that eliminates the ambiguity as to which region of space is included. This class acts as a proper surface, meaning that unary + and - operators applied to it will produce a half-space. The negative side is defined to be the region inside of the cone.
> 
> I feel like "ambiguity" surface here is a bit counterintuitive. Would "disambiguation" surface be better?

Based on the purpose described here, I would agree.

**Review by pshriwise — APPROVED** [review]
Thanks @MicahGale!

**Review by gridley — APPROVED** [review]
LGTM @MicahGale!

**Pull request merged**